lib/slack/channel.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Slack.Channel do
+  @moduledoc """
+  Functions for interacting with channels in Slack.
+  """
+  alias Slack.Lookups
+
+  defstruct [:id, :name]
+
+  @doc """
+  Returns a new `%Slack.Channel{}`.
+
+  opts - order doesn't matter
+    id - opaque id, as a string, of the channel
+    name - human readable name of channel
+  """
+  def new(opts) do
```

Comment:
I think I should have been more clear when commenting on this.

From what I understand about the API we should need only `slack` and a `channel_id`. I don't think it makes sense to support creating of new channels manually.

I think the API I can imagine being useful is that we pass a channel map and massage it into an actual channel. We also seem to be missing quite a few fields here according to the docs: https://api.slack.com/types/channel

What I meant by removing the error handling is that if we attempt to create a channel and fail we have some really bad state and Slack should crash. The crash gives it a chance to restart and get back into a good state.

Reply:
> we support both the web api and the RTM api. This change blurs that line quite a bit

Yeah, i implemented only the parts i needed. I was assuming that it would expand to cover the rest of that information at time passed.

Reply:
> I don't think it makes sense to support creating of new channels manually.

Fair. It might make testing a little more coupled to the struct shape but that is probably fine.

Reply:
> What I meant by removing the error handling is that if we attempt to create a channel and fail we have some really bad state and Slack should crash.

I think that is exactly what it was doing before. Raising an exception is very likely to crash the process. Much more likely, in fact, than returning an error tuple. I think the error handling (both now and earlier) is pretty good about not hiding errors. Before it caught them and then raised a better message not it detects errors and returns error tuples with better messages.

Reply:
I'm saying we should do 0 error handling and "let it crash"™. No try, no error tuple, just a good ole' exception caused by our logic and some bad data.

Reply:
I think that leads to code that is extremely hard to debug.